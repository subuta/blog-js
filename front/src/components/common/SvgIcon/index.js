import React from 'react'
import PropTypes from 'prop-types'

import withStyles from './style'

import { compose, setDisplayName, setPropTypes } from 'recompose'

const enhance = compose(
  withStyles,
  setDisplayName('SvgIcon'),
  setPropTypes({
    name: PropTypes.string
  })
)

export default enhance((props) => {
  const {
    name,
    className,
    styles,
    ...rest
  } = props

  let iconClass = `svg-icon ${styles.Icon}`
  if (className) {
    iconClass += ` ${className}`
  }

  return (
    <svg {...rest} className={iconClass}>
      <use xlinkHref={`/assets/symbol/svg/sprite.symbol.svg#${name}`} />
    </svg>
  )
})
