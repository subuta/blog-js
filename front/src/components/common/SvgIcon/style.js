import {
  registerStyles
} from 'src/utils/style'

import {
  BLACK_COLOR,
  PRIMARY_COLOR
} from 'src/constants/style'

const Icon = {
  display: 'inline-block',
  height: '100%',
  width: '100%',
  color: 'inherit'
}

export default registerStyles({
  Icon
})
