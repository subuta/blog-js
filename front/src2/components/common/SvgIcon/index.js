import React from 'react'
import PropTypes from 'prop-types'

import classes from './style'

import { compose, setDisplayName, setPropTypes } from 'recompose'

const enhance = compose(
  setDisplayName('SvgIcon'),
  setPropTypes({
    name: PropTypes.string
  })
)

export default enhance((props) => {
  const {
    name,
    className,
    ...rest
  } = props

  let iconClass = `svg-icon ${classes.Icon}`
  if (className) {
    iconClass += ` ${className}`
  }

  return (
    <svg {...rest} className={iconClass}>
      <use xlinkHref={`/assets/symbol/svg/sprite.symbol.svg#${name}`} />
    </svg>
  )
})
