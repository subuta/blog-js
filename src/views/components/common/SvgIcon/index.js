import React from 'react'
import PropTypes from 'prop-types'

import { staticFolder } from 'src/views/constants/config'
import withStyles from './style'

import { compose, setDisplayName, setPropTypes } from 'recompose'

const enhance = compose(
  withStyles,
  setDisplayName('SvgIcon'),
  setPropTypes({
    name: PropTypes.string
  })
)

export default enhance((props) => {
  const {
    name,
    className,
    styles,
    ...rest
  } = props

  let iconClass = `svg-icon svg-${name}-dims ${styles.Icon}`
  if (className) {
    iconClass += ` ${className}`
  }

  return (
    <svg {...rest} className={iconClass}>
      <use xlinkHref={`${staticFolder}/assets/symbol/svg/sprite.symbol.svg#${name}`}/>
    </svg>
  )
})
