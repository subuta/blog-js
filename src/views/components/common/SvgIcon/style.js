import createWithStyles from 'src/views/utils/style'

import {
  BLACK_COLOR,
  PRIMARY_COLOR
} from 'src/views/constants/style'

const Icon = {
  display: 'inline-block',
  height: '100%',
  width: '100%',
  color: 'inherit',
  verticalAlign: 'top'
}

export default createWithStyles({
  Icon
})
