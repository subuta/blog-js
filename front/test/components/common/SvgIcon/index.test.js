/* global describe, it */
import React from 'react'
import { create } from 'react-test-renderer'

import SvgIcon from 'src/components/common/SvgIcon'

describe('SvgIcon', () => {
  it('should be rendered correctly', () => {
    const tree = create(
      <SvgIcon
        className="logo"
        name="logo"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
