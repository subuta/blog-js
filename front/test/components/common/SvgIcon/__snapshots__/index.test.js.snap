// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon should be rendered correctly 1`] = `
<svg
  className="svg-icon f10ljuk0 logo"
>
  <use
    xlinkHref="/assets/symbol/svg/sprite.symbol.svg#logo"
  />
</svg>
`;
